velox/functions/prestosql/DecimalArithmetic.cpp

```diff
@@ -188,27 +188,35 @@ class Multiply {
 
 std::vector<std::shared_ptr<exec::FunctionSignature>>
 decimalMultiplySignature() {
-  return {exec::FunctionSignatureBuilder()
-              .returnType("DECIMAL(r_precision, r_scale)")
-              .argumentType("DECIMAL(a_precision, a_scale)")
-              .argumentType("DECIMAL(b_precision, b_scale)")
-              .variableConstraint(
-                  "r_precision", "min(38, a_precision + b_precision)")
-              .variableConstraint("r_scale", "a_scale + b_scale")
-              .build()};
+  return {
+      exec::FunctionSignatureBuilder()
+          .integerVariable("a_precision")
+          .integerVariable("a_scale")
+          .integerVariable("b_precision")
+          .integerVariable("b_scale")
+          .integerVariable("r_precision", "min(38, a_precision + b_precision)")
+          .integerVariable("r_scale", "a_scale + b_scale")
+          .returnType("DECIMAL(r_precision, r_scale)")
+          .argumentType("DECIMAL(a_precision, a_scale)")
+          .argumentType("DECIMAL(b_precision, b_scale)")
+          .build()};
 }
 
 std::vector<std::shared_ptr<exec::FunctionSignature>>
 decimalAddSubtractSignature() {
   return {
       exec::FunctionSignatureBuilder()
+          .integerVariable("a_precision")
```

Comment:
Do you think combing integerVariable into the argumentType call. Similar to what JAVA does:

`parseTypeSignature("decimal(a_precision, a_scale)", ImmutableSet.of("a_precision", "a_scale"));`
Looks a bit concise.

Reply:
The downside of combining is that we cannot reuse variables. Also, we have to explicitly specify if it is an integer or a type parameter.